src/cli/commands/upgrade.js

```diff
@@ -19,6 +19,20 @@ export async function run(
   args: Array<string>,
 ): Promise<void> {
   const lockfile = args.length ? await Lockfile.fromDirectory(config.cwd, reporter) : new Lockfile();
-  const install = new Add(args, flags, config, reporter, lockfile);
+  let addArgs = [...args];
+
+  if (args.length) {
+    const [dependency] = args;
+    const manifest = await config.readRootManifest() || {};
+    const dependencies = manifest.dependencies || {};
+    const remoteSource = dependencies[dependency];
+
+    if (remoteSource && remoteSource.includes(':')) {
```

Comment:
My first thought is that this check should probably be for the `remoteSource` matching any "exotic" resolver (git, bitbucket, file, tarball, etc...).

Might look at `package-request.js` which contains the function

```
static getExoticResolver(pattern: string): ?Function
```